src/HttpHeader.cc

```diff
@@ -747,6 +748,22 @@ HttpHeader::refreshMask()
     }
 }
 
+bool
+HttpHeader::tooLarge() const
+{
+    size_t max = 0;
+    switch (owner) {
+    case hoRequest:
+        max = Config.maxRequestHeaderSize;
+    case hoReply:
+         max = Config.maxReplyHeaderSize;
+    default:
+        return false;
+    }
```

Comment:
What happens when received hoErrorDetail is bigger than Config.maxReplyHeaderSize? If that causes no problems, let's add the corresponding dedicated `case` that returns false, with a comment justifying such "ignorance".